src/scripts/sniperino.ts

```diff
@@ -0,0 +1,101 @@
+// Description:
+//  Beat the sniperino
+//  
+//
+// Commands:
+//  sniperino( me)? - start a new game of sniperino 
+//  snipe( me)? - roll a snipe roll in an attempt to beat the sniperino
+//
+// Author:
+//  Steve Shipsey
+
+import { Robot } from "tsbot";
+
+interface ISniper {
+    name: string;
+    gamesPlayed: number;
+    gamesWon: number;
+    currentSnipe?: number;
+
+    getWinRate(): number;
+    getNewSnipe();
+}
+
+class Sniper implements ISniper {
+    public name: string;
+    public gamesPlayed: number;
+    public gamesWon: number;
+    public currentSnipe: number;
+    
+    constructor(n: string) {
+        this.name = n.slice(0, n.indexOf(" "));
```

Comment:
The full name (not actually unique, but more unique than first name) was to be used as the key for the brain dictionary, but I was storing the name property within the object as just the first name since that's the only part of the name the script would ever reply with.